File: client2/src/app/shared/models/admin/selectionDecision.ts
export interface ISelectionDecision
{
     checked: boolean;
     id: number;
     cVRefId: number;
     orderItemId: number;
     categoryId: number;
     categoryRef: string;
     orderNo: number;
     customerName: string;
     applicationNo: number;
     candidateId: number;
     candidateName: string;
     decisionDate: Date;
     selectionStatusId: number;
     selectedOn: number;
     employment: IEmployment;
     remarks: string;
}

export interface IEmployment
{
     cVRefId: number;
     selectionDecisionId: number;
     selectedOn: Date;
     charges: number;
     salaryCurrency: string;
     salary: number;
     contractPeriodInMonths: number;
     weeklyHours: number;
     housingProvidedFree: boolean;
     housingAllowance: number;
     foodProvidedFree: boolean;
     foodAlowance: number;
     transportProvidedFree: boolean;
     transportAllowance: number;
     otherAllowance: number;
     leavePerYearInDays: number;
     leaveAirfareEntitlementAfterMonths: number;
     offerAcceptedOn: Date;
     //new additions
     categoryId: number;
     categoryName: string;
     orderItemId: number;
     orderId: number;
     orderNo: number;
     customerId: number;
     customerName: string;
     candidateId: number;
     applicationNo: number;
     candidateName: string;
     companyName: string;     //agent name
     remarks: string;
}

File: core/Entities/HR/SelectionDecision.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using core.Entities.Orders;

namespace core.Entities.HR
{
    public class SelectionDecision: BaseEntity
    {
          public SelectionDecision()
          {
          }

        public SelectionDecision(  string categoryName, int orderId, int orderNo, string customerName, 
            int applicationNo, string candidateName, DateTime decisionDate, 
              int selectionStatusId, string remarks, CVRef cvRef)
           {
               CategoryName = categoryName;
               OrderId = orderId;
               OrderNo = orderNo;
               CustomerName = customerName;
               ApplicationNo = applicationNo;
               CandidateName = candidateName;
               DecisionDate = decisionDate;
               SelectionStatusId = selectionStatusId;
               Remarks = remarks;
               CVRef = cvRef;
          }

        public SelectionDecision(int categoryId, string categoryName, int orderId, int orderNo, string customerName, 
            int applicationNo, string candidateName, DateTime decisionDate, int selectionStatusId, string remarks, 
            int orderitemid, int candidateid,DateTime selectedon,
            int charges, CVRef cvRef, Employment employment)
           {
               //CVRefId = cVRefId;
               OrderItemId = orderitemid;
               CategoryId = categoryId;
               CategoryName = categoryName;
               OrderId = orderId;
               OrderNo = orderNo;
               CustomerName = customerName;
               ApplicationNo = applicationNo;
               CandidateId = candidateid;
               CandidateName = candidateName;
               DecisionDate = decisionDate;
               SelectionStatusId = selectionStatusId;
               Remarks = remarks;
               Charges=charges;
               CVRef = cvRef;
               Employment = employment;
          }

          public int CVRefId { get; set; }
          public int OrderItemId { get; set; }
          public int CategoryId { get; set; }
          public string CategoryName {get; set;}
          public int OrderId { get; set; }
          public int OrderNo {get; set;}
          public string CustomerName {get; set;}
          public int ApplicationNo {get; set;}
          public int CandidateId {get; set;}
          public string CandidateName {get; set;}
          public DateTime DecisionDate {get; set;}
          public int SelectionStatusId { get; set; }
          [ForeignKey("CVRefId")]
          public virtual CVRef CVRef {get; set;}
          public DateTime SelectedOn { get; set; }
          public int Charges {get; set;}
          public string Remarks {get; set;}
          #nullable enable
          public Employment? Employment {get; set;}
    }
}